test/tests/neml2/tests

```diff
@@ -8,7 +8,7 @@
     requirement = 'BlackBear shall be capable of capturing errors occurring during the evaluation of a NEML2 model.'
     expect_err = 'To debug NEML2 related issues'
     collections = 'SYSTEM'
-    required_objects = 'CauchyStressFromNEML2'
+    required_submodule = 'contrib/neml2'
```

Comment:
I'm all for this if we can unstuck next. Though in the future we shouldn't be relying on contrib/neml2 because a user may build blackbear with an out of source NEML_DIR.

Reply:
Yep, I definitely agree. the commit message alludes to this:

> This works pretty poorly when you have NEML2 out of tree, but
it's good enough for now. Eventually we should be able to get
configure options and flags into the test harness